tachyon/zk/r1cs/constraint_matrices.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 arkworks contributors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.arkworks and the LICENCE-APACHE.arkworks
+// file.
+
+#ifndef TACHYON_ZK_R1CS_CONSTRAINT_MATRICES_H_
+#define TACHYON_ZK_R1CS_CONSTRAINT_MATRICES_H_
+
+#include <stddef.h>
+
+#include <string>
+
+#include "absl/strings/substitute.h"
+
+#include "tachyon/zk/r1cs/matrix.h"
+
+namespace tachyon::zk::r1cs {
+
+// The A, B and C matrices of a Rank-One |ConstraintSystem|.
+// Also contains metadata on the structure of the constraint system
+// and the matrices.
```

Comment:
Could you add a NOTE or TODO or FIXME comment here that you want to separate the current A, B, C matrices with the solution vectors?